flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/ZooKeeperCompletedCheckpointStore.java

```diff
@@ -107,11 +108,27 @@
 	 * @throws Exception
 	 */
 	public ZooKeeperCompletedCheckpointStore(
-			int maxNumberOfCheckpointsToRetain,
-			CuratorFramework client,
-			String checkpointsPath,
-			RetrievableStateStorageHelper<CompletedCheckpoint> stateStorage,
-			Executor executor) throws Exception {
+		int maxNumberOfCheckpointsToRetain,
+		CuratorFramework client,
+		String checkpointsPath,
+		RetrievableStateStorageHelper<CompletedCheckpoint> stateStorage,
+		Executor executor) throws Exception {
+		this(maxNumberOfCheckpointsToRetain,
+			client,
+			checkpointsPath,
+			stateStorage,
+			executor,
+			null);
```

Comment:
we could pass `new ZooKeeperStateHandleStore<>(this.client, stateStorage)` instead; then we wouldn't need the null check in the other constructor

Reply:
we don't initalize `this.client` in this constructor. and before construct `checkpointsInZooKeeper`, we need to adapt the client by 

```java
        // Ensure that the checkpoints path exists
		client.newNamespaceAwareEnsurePath(checkpointsPath)
				.ensure(client.getZookeeperClient());

		// All operations will have the path as root
		this.client = client.usingNamespace(client.getNamespace() + checkpointsPath);
```

calling `this(...)` should be the first statement, I don't know how to do some initialization first and then call another constructor.

Reply:
@zentol I try to get rid of the null check by adapting the client first. However, this introduces two more `private` constructors for the unified path. I'd like to hear from your advice on this change.